operator/v1/types_ingress.go

```diff
@@ -410,6 +410,33 @@ type AWSNetworkLoadBalancerParameters struct {
 // HostNetworkStrategy holds parameters for the HostNetwork endpoint publishing
 // strategy.
 type HostNetworkStrategy struct {
+	// protocol specifies whether the IngressController expects incoming
+	// connections to use plain TCP or whether the IngressController expects
+	// PROXY protocol.
+	//
+	// PROXY protocol can be used with load balancers that support it to
+	// communicate the source addresses of client connections when
+	// forwarding those connections to the IngressController.  Using PROXY
+	// protocol enables the IngressController to report those source
+	// addresses instead of reporting the load balancer's address in HTTP
+	// headers and logs.  Note that enabling PROXY protocol on the
+	// IngressController will cause connections to fail if you are not using
+	// a load balancer that uses PROXY protocol to forward connections to
+	// the IngressController.  See
+	// http://www.haproxy.org/download/2.2/doc/proxy-protocol.txt for
+	// information about PROXY protocol.
+	//
+	// The following values are valid for this field:
+	//
+	// * The empty string.
+	// * "TCP".
+	// * "PROXY".
+	//
+	// By default, TCP is used without PROXY protocol.
```

Comment:
Can you indicate that defaults may change over time?  These seem unlikely to change, but I'd like to make clear that empty means, "no opinion, give me the current default"

Same below.